src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VcfReader.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.io.readers;
+
+import org.opencb.biodata.formats.variant.vcf4.io.VariantVcfReader;
+import org.opencb.biodata.models.variant.Variant;
+import org.opencb.biodata.models.variant.VariantSource;
+import org.springframework.batch.item.file.FlatFileItemReader;
+import org.springframework.core.io.Resource;
+import uk.ac.ebi.eva.pipeline.io.GzipLazyResource;
+import uk.ac.ebi.eva.pipeline.io.mappers.VcfLineMapper;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * VCF file reader.
+ *
+ * This Reader uses a {@link VcfLineMapper} to parse each line, and {@link VariantVcfReader} to fill the VariantSource.
+ *
+ * @author Jose Miguel Mut Lopez &lt;jmmut@ebi.ac.uk&gt;
+ */
+public class VcfReader extends FlatFileItemReader<List<Variant>> {
+    private final VariantSource source;
+    private final File file;
+
+    public VcfReader(VariantSource source, File file) {
+        this.source = source;
+        this.file = file;
+        Resource resource = new GzipLazyResource(file);
```

Comment:
Maybe the system should detect whether the file is compressed or not and create a resource accordingly.